packages/devtools-reps/src/reps/tests/grip-map-entry.js

```diff
@@ -0,0 +1,143 @@
+  it("return the expected object", () => {
+    const entry = createGripMapEntry("A", 0);
+    expect(entry).toEqual(stubs.get("A → 0"));
+  });
+});
+
+describe("GripMapEntry - complex", () => {
+  it.only("Handles complex objects as key and value", () => {
+    let stub = gripArrayStubs.get("testBasic");
+    let entry = createGripMapEntry("A", stub);
+    expect(renderRep(entry, MODE.TINY).text()).toEqual("A → []");
+    expect(renderRep(entry, MODE.SHORT).text()).toEqual("A → Array []");
+    expect(renderRep(entry, MODE.LONG).text()).toEqual("A → Array []");
+
+    entry = createGripMapEntry(stub, "A");
+    expect(renderRep(entry, MODE.TINY).text()).toEqual(`[] → "A"`);
+    expect(renderRep(entry, MODE.SHORT).text()).toEqual(`Array [] → "A"`);
+    expect(renderRep(entry, MODE.LONG).text()).toEqual(`Array [] → "A"`);
+
+    stub = gripArrayStubs.get("testMaxProps");
+    entry = createGripMapEntry("A", stub);
+    expect(renderRep(entry, MODE.TINY).text()).toEqual(`A → […]`);
+    expect(renderRep(entry, MODE.SHORT).text()).toEqual(`A → Array [ 1, "foo", {} ]`);
+    expect(renderRep(entry, MODE.LONG).text()).toEqual(`A → Array [ 1, "foo", {} ]`);
+
+    entry = createGripMapEntry(stub, "A");
+    expect(renderRep(entry, MODE.TINY).text()).toEqual(`[…] → "A"`);
+    expect(renderRep(entry, MODE.SHORT).text()).toEqual(`Array [ 1, "foo", {} ] → "A"`);
+    expect(renderRep(entry, MODE.LONG).text()).toEqual(`Array [ 1, "foo", {} ] → "A"`);
+
+    stub = gripArrayStubs.get("testMoreThanShortMaxProps");
+    entry = createGripMapEntry("A", stub);
+    expect(renderRep(entry, MODE.TINY).text()).toEqual(`A → […]`);
+    expect(renderRep(entry, MODE.SHORT).text())
+      .toEqual(`A → Array [ "test string", "test string", "test string", … ]`);
+    expect(renderRep(entry, MODE.LONG).text()).toEqual(
+      `A → Array [ "test string", "test string", "test string", "test string" ]`);
+
+    entry = createGripMapEntry(stub, "A");
+    expect(renderRep(entry, MODE.TINY).text()).toEqual(`[…] → "A"`);
+    expect(renderRep(entry, MODE.SHORT).text())
+      .toEqual(`Array [ "test string", "test string", "test string", … ] → "A"`);
+    expect(renderRep(entry, MODE.LONG).text()).toEqual(
+      `Array [ "test string", "test string", "test string", "test string" ] → "A"`);
+  });
+
+  it("Handles Element Nodes as key and value", () => {
+    const stub = nodeStubs.get("Node");
+
+    const onInspectIconClick = jest.fn();
+    const onDOMNodeMouseOut = jest.fn();
+    const onDOMNodeMouseOver = jest.fn();
+
+    let entry = createGripMapEntry("A", stub);
+    let renderedComponent = renderRep(entry, MODE.TINY, {
+      onInspectIconClick,
+      onDOMNodeMouseOut,
+      onDOMNodeMouseOver,
+    });
+    expect(renderRep(entry, MODE.TINY).text())
+      .toEqual("A → input#newtab-customize-button.bar.baz");
+
+    let node = renderedComponent.find(".objectBox-node");
+    let icon = node.find(".open-inspector");
```

Comment:
For inspect icon tests it'd be nice if we didn't need to retest the whole interaction for every rep that can have a  PropRep as a child. Either some kind of shared testing utility that does this event simulation / mocked calls dance, or if we could assume that the inspect icon works fine as long as the `.open-inspector` DOM Node is there (as I'm sure it's tested on its own)